omega_moderne_client/campaign/campaign_executor.py

```diff
@@ -0,0 +1,67 @@
+import time
+from dataclasses import dataclass
+from typing import Any, List
+
+from .campaign import Campaign
+from omega_moderne_client.client.gpg_key_config import GpgKeyConfig
+from omega_moderne_client.client.moderne_client import ModerneClient
+
+
+@dataclass(frozen=True)
+class CampaignExecutor:
+    campaign: Campaign
+    client: ModerneClient
+
+    def execute_recipe_and_await(
+            self,
+            target_organization_id="Default"
+    ) -> 'RecipeExecutionResult':
+        print(f"Running campaign {self.campaign.name}...")
+        run_id = self.client.run_campaign(
+            self.campaign,
+            target_organization_id=target_organization_id
+        )
+        print(f"Waiting for recipe run {run_id} to complete...")
+        print(f"View live on Moderne https://{self.client.domain}/results/{run_id}")
+
+        while True:
+            state = self.client.query_recipe_run_status(run_id)
+            print(f"Recipe {run_id} state: {state}")
+            if state == "FINISHED":
```

Comment:
Would there be a need to record a state that paused or failed in the middle? I'd presumed it'd fall under the cancel state, but leaves the questions (for auditing) who/how was it canceled.

Reply:
Probably. I'm not sure if it says who canceled it